docs/modules/ROOT/pages/documentation/user-guides/starter-with-jpa.adoc

```diff
@@ -0,0 +1,101 @@
+
+== MySQL based application with an external definition of the DataSource.
+
+1) Start by generating a project using the MicroProfile Starter website. You can use the Command Line to create a project for Payara Micro without any examples of the MicroProfile specifications.
+
+[source, shell]
+----
+curl -O -J 'https://start.microprofile.io/api/project?supportedServer=PAYARA_MICRO'
+----
+
+2) Add the Jakarta Web profile maven dependencies to the project to have access to the JPA API. Set the scope to _provided_ as all required dependencies are already bundled within Payara Micro. 
+
+[source, xml]
+----
+<dependency>
+    <groupId>jakarta.platform</groupId>
+    <artifactId>jakarta.jakartaee-web-api</artifactId>
+    <version>8.0.0</version>
+    <scope>provided</scope>
+</dependency>
+----
+
+3) Add the persistence.xml file to define the Persistence unit. Create a file src/main/resources/META-INF/persistence.xml wit the following content
+
+[source, xml]
+----
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<persistence xmlns="http://xmlns.jcp.org/xml/ns/persistence" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="2.1" xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd">
+    <persistence-unit name="demoPU" transaction-type="JTA">
+        <jta-data-source>jdbc/local-db</jta-data-source>
+    </persistence-unit>
+</persistence>
+----
+
+Important to note here is the JNDI name of the DataSource we need to create in one of the following steps. 
+
+4) Define the JPA Entities and use @PersistenceContext just as you would do when using JPA in any Java EE or Jakarta EE compliant runtime.
+
+5) Create a post-boot command file that will create the DataSource when Payara Micro starts up. The file name and location is of no importance but needs to be referenced when launching the application.
+
+[source, shell]
+----
+create-jdbc-connection-pool --datasourceclassname com.mysql.jdbc.jdbc2.optional.MysqlDataSource --restype javax.sql.DataSource demo-pool
+set resources.jdbc-connection-pool.demo-pool.property.password=password resources.jdbc-connection-pool.demo-pool.property.databaseName=fishes resources.jdbc-connection-pool.demo-pool.property.serverName=localhost resources.jdbc-connection-pool.demo-pool.property.user=demo resources.jdbc-connection-pool.demo-pool.property.portNumber=3306
+
+create-jdbc-resource --enabled=true --poolName=demo-pool --target=domain jdbc/local-db
+create-resource-ref --enabled=true --target=server jdbc/local-db
+----
+
+Important here is the database information like Server name, user name, and password. They can be specified as variables so that you can define them with JVM System parameters, environment variables, or Kubernetes Secrets, but there is no need to do this since they are defined in this external file which can environment specific and not registered in the Source repository.
```

Comment:
```suggestion
Important here is the database information like Server name, user name, and password. They can be specified as variables so that you can define them with JVM System parameters, environment variables, or Kubernetes Secrets, but there is no need to do this since they are defined in this external file which is already environment specific and not registered in the Source repository.
```